dauphine-open-data-lib/src/main/java/io/github/oliviercailloux/y2018/opendata/entity/Person.java

```diff
@@ -1,5 +1,312 @@
-package io.github.oliviercailloux.y2018.opendata.entity;
-
-public class Person {
-
-}
+package io.github.oliviercailloux.y2018.opendata.entity;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.persistence.CollectionTable;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.NotNull;
+import com.google.common.base.Strings;
+
+/**
+ * Provides a basic implementation of class Person.<br />
+ * Person is either Student or Teacher. In this class you have basic attributes
+ * to manipulate class Person and getters an setters. In our code INE is the
+ * acronym of National Student Identification All our getters should not return
+ * null string All our method get for a List are protected from unsafe
+ * modification we handle the adding of null value in a list by making mandatory
+ * parameter of method modifying a list <br />
+ * 
+ * @author Gandi Taric
+ */
+
+@Entity
+public class Person extends AbstractEntity {
+
+	private static final long serialVersionUID = -4768931293875213592L;
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+
+	/**
+	 * Firstname field mandatory
+	 */
+	@NotNull
+	private String firstName;
+
+	/**
+	 * This fiel is also madatory all person mus have a lastname
+	 */
+	@NotNull
+	private String LastName;
+
+	/**
+	 * This will help to distinguish student from teachers and is mandatory
+	 */
+	@NotNull
+	private String accountType;
+
+	/**
+	 * This field represent National Student Identification code in case person is a
+	 * student it will be filled with empty string if person is a teacher
+	 */
+	private String ine;
+
+	/**
+	 * This field help to check if person object is active or not
+	 */
+	private boolean isActive;
+
+	/**
+	 * This represent teacher's office when person is teacher
+	 */
+	private String office;
+
+	/**
+	 * This represent the phone number of the person
+	 */
+	private String phoneNumer;
+
+	/**
+	 * this field represent the training of the student and will be set to empty
+	 * string if it is a teacher
+	 */
+	private String training;
+
+	/**
+	 * This represent fax of the person
+	 */
+	private String fax;
+
+	/**
+	 * this field represent the personnal mail of the person.a person can have many
+	 * personal mail
+	 */
+	@Column(nullable = false)
+	@ElementCollection
+	@CollectionTable(name = "personnalMail")
+	private List<String> personnalMail;
+
+	/**
+	 * a person in this context have mandatory dauphine mail so it can't be null
```

Comment:
Can’t be empty, vous voulez dire, j’imagine ? Une telle liste ne devrait de toute façon jamais être nulle (même dans le cas de personalMail par exemple).

Reply:
@oliviercailloux 
Effectivement. je voulais dire empty.
Commentaire corrigé.